From 513a0f9f7e1e3e6fdeb958872a2cd96d1b25ba65 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Mon, 11 Dec 2023 12:54:34 +0100
Subject: [PATCH 431/841] feat(stm32mp2-fdts): add nvram (NVMEM) node under
 tamp in stm32mp251.dtsi

Backup Registers nvram will be accessed using the NVMEM framework

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 269459aa9cf132ca9de8c01a40adbdc3673c3673 feat(stm32mp2-fdts): add nvram (NVMEM) node under tamp in stm32mp251.dtsi]
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
Change-Id: I995db2915bc1a085a9c32778b8381cd36897789e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/343945
Tested-by: Simeon MARIJON <simeon.marijon@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Simeon MARIJON <simeon.marijon@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 fdts/stm32mp251.dtsi | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index 1ab142b3f..5a9653637 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -430,6 +430,35 @@
 			reg = <0x44230000 0x10000>;
 		};
 
+		tamp: tamp@46010000 {
+			compatible = "st,stm32mp25-tamp";
+			reg = <0x46010000 0x400>;
+			clocks = <&rcc CK_BUS_RTC>;
+			interrupts = <GIC_SPI 14 IRQ_TYPE_LEVEL_HIGH>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges;
+
+			nvram: nvram@46010100 {
+				compatible = "st,stm32mp25-tamp-nvram";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				reg = <0x46010100 0x200>;
+
+				stop2_entrypoint: tamp-bkp@2c {
+					reg = <0x2c 0x4>;
+				};
+				fwu_info: tamp-bkp@c0 {
+					/* see firmware update info feature */
+					reg = <0xc0 0x4>;
+				};
+				boot_mode: tamp-bkp@180 {
+					reg = <0x180 0x4>;
+				};
+
+			};
+		};
+
 		ddr: ddr@48040000 {
 			compatible = "st,stm32mp2-ddr";
 			reg = <0x48040000 0x10000>,
-- 
2.47.2

